cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -46,8 +72,14 @@ private[lagom] object JoinClusterImpl {
           // since that would have blocked the shutdown of the ActorSystem.
           val t = new Thread(new Runnable {
             override def run(): Unit = {
-              if (Try(Await.ready(system.whenTerminated, 10.seconds)).isFailure)
-                System.exit(-1)
+              if (Try(Await.ready(system.whenTerminated, 10.seconds)).isFailure) {
+                println("Halting JVM.")
+                Runtime.getRuntime.halt(EXIT_TIMEOUT_WITH_HALT)
+              }
+              if (needsExiting) {
+                println("Proceed to JVM shutdown with exit status: " + exitStatus.get())
+                System.exit(EXIT_TIMEOUT)
+              }
             }
```

Comment:
If we remove the `if` block from line 40 I think we should still keep the approach here to try to `System.exit` and, if there's a timeout, `halt`.

Reply:
Invoking `System.exit` with a 10s fallback to a `Runtime.halt` provides a fast shutdown of up to 10s.

Reply:
BTW: it's important to _not_ use logging (and instead use `println`)  because during the shutdown loggers are, well, shut down (d'uh!) and we'd miss the traces.

Reply:
That makes sense, but I wonder whether it would be better to write to stderr